flink-staging/flink-streaming/flink-storm-compatibility/src/main/java/org/apache/flink/stormcompatibility/api/FlinkClient.java

```diff
@@ -0,0 +1,321 @@
+import org.apache.flink.configuration.ConfigConstants;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.configuration.GlobalConfiguration;
+import org.apache.flink.core.fs.Path;
+import org.apache.flink.runtime.akka.AkkaUtils;
+import org.apache.flink.runtime.client.JobStatusMessage;
+import org.apache.flink.runtime.jobgraph.JobGraph;
+import org.apache.flink.runtime.jobmanager.JobManager;
+import org.apache.flink.runtime.messages.JobManagerMessages;
+import org.apache.flink.runtime.messages.JobManagerMessages.CancelJob;
+import org.apache.flink.runtime.messages.JobManagerMessages.RunningJobsStatus;
+
+import scala.Some;
+import scala.concurrent.Await;
+import scala.concurrent.Future;
+import scala.concurrent.duration.FiniteDuration;
+import akka.actor.ActorRef;
+import akka.actor.ActorSystem;
+import akka.pattern.Patterns;
+import akka.util.Timeout;
+import backtype.storm.Config;
+import backtype.storm.generated.AlreadyAliveException;
+import backtype.storm.generated.InvalidTopologyException;
+import backtype.storm.generated.KillOptions;
+import backtype.storm.generated.Nimbus;
+import backtype.storm.generated.NotAliveException;
+import backtype.storm.generated.SubmitOptions;
+import backtype.storm.utils.NimbusClient;
+import backtype.storm.utils.Utils;
+
+
+
+
+
+/**
+ * {@link FlinkClient} mimics a Storm {@link NimbusClient} and {@link Nimbus}{@code .Client} at once, to interact with
+ * Flink's JobManager instead of Storm's Nimbus.
+ */
+public class FlinkClient {
+	/**
+	 * The jobmanager's host name.
+	 */
+	private final String jobManagerHost;
+	/**
+	 * The jobmanager's rpc port.
+	 */
+	private final int jobManagerPort;
+	/**
+	 * The user specified timeout in milliseconds.
+	 */
+	private final String timeout;
+	
+	
+	
+	/*
+	 * The following methods are derived from "backtype.storm.utils.NimbusClient"
+	 */
+	
+	/**
+	 * Instantiates a new {@link FlinkClient} for the given configuration, host name, and port. If values for
+	 * {@link Config#NIMBUS_HOST} and {@link Config#NIMBUS_THRIFT_PORT} of the given configuration are ignored.
+	 * 
+	 * @param conf
+	 *            A configuration.
+	 * @param host
+	 *            The jobmanager's host name.
+	 * @param port
+	 *            The jobmanager's rpc port.
+	 */
+	public FlinkClient(final Map<?, ?> conf, final String host, final int port) {
+		this(conf, host, port, null);
+	}
+	
+	/**
+	 * Instantiates a new {@link FlinkClient} for the given configuration, host name, and port. If values for
+	 * {@link Config#NIMBUS_HOST} and {@link Config#NIMBUS_THRIFT_PORT} of the given configuration are ignored.
+	 * 
+	 * @param conf
+	 *            A configuration.
+	 * @param host
+	 *            The jobmanager's host name.
+	 * @param port
+	 *            The jobmanager's rpc port.
+	 * @param timeout
+	 */
+	public FlinkClient(final Map<?, ?> conf, final String host, final int port, final Integer timeout) {
```

Comment:
FlinkClient offers the unified interface as NimbusClient and Nimbus.Client (in the compatibility layer, it does not make sense to destingues both). Thus, the constructor offers the same parameters (even in not used right now). The same holds for Map -- Storm does not use any generics.